internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
Why not continue using the env on testWorkflowEnvironment and let the default be the global GetHostEnvironment() and allow someone to override it if required? This will allow us to test with isolated host environments.